io/plugins/eu.esdihumboldt.hale.io.jdbc.test/src/eu/esdihumboldt/hale/io/jdbc/test/AbstractDBTest.java

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (c) 2015 Data Harmonisation Panel
+ * 
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution. If not, see <http://www.gnu.org/licenses/>.
+ * 
+ * Contributors:
+ *     Data Harmonisation Panel <http://www.dhpanel.eu>
+ */
+
+package eu.esdihumboldt.hale.io.jdbc.test;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.net.URI;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.List;
+
+import com.typesafe.config.Config;
+
+import eu.esdihumboldt.hale.common.core.io.Value;
+import eu.esdihumboldt.hale.common.core.io.report.IOReport;
+import eu.esdihumboldt.hale.common.core.io.supplier.NoStreamInputSupplier;
+import eu.esdihumboldt.hale.common.core.io.supplier.NoStreamOutputSupplier;
+import eu.esdihumboldt.hale.common.instance.model.Instance;
+import eu.esdihumboldt.hale.common.instance.model.InstanceCollection;
+import eu.esdihumboldt.hale.common.instance.model.InstanceUtil;
+import eu.esdihumboldt.hale.common.instance.model.ResourceIterator;
+import eu.esdihumboldt.hale.common.instance.model.TypeFilter;
+import eu.esdihumboldt.hale.common.instance.model.impl.DefaultInstanceCollection;
+import eu.esdihumboldt.hale.common.schema.model.Schema;
+import eu.esdihumboldt.hale.common.schema.model.TypeDefinition;
+import eu.esdihumboldt.hale.common.schema.model.impl.DefaultSchemaSpace;
+import eu.esdihumboldt.hale.io.jdbc.JDBCConnection;
+import eu.esdihumboldt.hale.io.jdbc.JDBCInstanceReader;
+import eu.esdihumboldt.hale.io.jdbc.JDBCInstanceWriter;
+import eu.esdihumboldt.hale.io.jdbc.JDBCSchemaReader;
+
+/**
+ * Base class for database tests.
+ * 
+ * @author Simon Templer
+ */
+public abstract class AbstractDBTest {
+
+	public static final String DB_UPTIME = ".dbUPTime";
+	private DBImageParameters dbi;
+	private DBDockerClient tdc;
+	public static final String ORACLE_DB = "ORCL";
+
+	/**
+	 * Setup host and database.
+	 */
+	public void setupDB(DBImageParameters dbi) {
```

Comment:
Why not pass the DBImageParameters in the test constructor?
Then you would not have to create a specific setupDB method in every test, but could define it in AbstractDBTest (including the Before annotation)